test/files/run/repl-completions.check

```diff
@@ -0,0 +1,28 @@
+
+scala> // completions!
+
+scala> :completions "foo".com
+[completions] combinations
+[completions] companion
+[completions] compare
+[completions] compareTo
+[completions] compareToIgnoreCase
+[completions] compose
```

Comment:
The REPL completions will offer completions that don't match the prefix, e.g.

```
scala> this.getClass.dM<TAB>
<console>:12: error: value dM is not a member of Class[T]
       this.getClass.dM<TAB>
                     ^
<console>:12: error: not found: value TAB
       this.getClass.dM<TAB>

scala> this.getClass.getDeclaredMethod
getDeclaredMethod   getDeclaredMethods
```

Is this likely to be a problem for the code in Emacs that consumes the list?

Reply:
It might also be useful to have `:completions  -all <code>`, which avoided any filtering of the result list, leaving that has a job for the IDE.

That would require a small refactor to let the caller to `ReplCompletions.complete` pass in the filtering lambda [passed to `matchingResults`](https://github.com/scala/scala/blob/92affd3e76d1a3019f1c35ac20af32360d914641/src/repl/scala/tools/nsc/interpreter/PresentationCompilerCompleter.scala#L85)

Reply:
completing `this.getClass.dM` to `this.getClass.getDeclaredMethod` will not pose a problem on emacs side, and actually might be even nice. However, I need to change the `completions` command to return the whole line, so I would like then this to return

```
scala> :completions this.getClass.dM
[completions] this.getClass.getDeclaredMethod
[completions] this.getClass.getDeclaredMethods
```

With this I can then on emacs-side include or filter out the completions that match the original line based on the users settings.

Reply:
@retronym any pointers where I should make the change to do this?